openpype/hosts/blender/plugins/create/create_render.py

```diff
@@ -0,0 +1,302 @@
+
+                output.file_slots.new(filepath)
+
+                aov_file_products.append(
+                    (render_pass.name, os.path.join(output_path, filepath)))
+
+            node_input = output.inputs[-1]
+
+            tree.links.new(render_pass, node_input)
+
+        return aov_file_products
+
+    @staticmethod
+    def imprint_render_settings(node, data):
+        RENDER_DATA = "render_data"
+        if not node.get(RENDER_DATA):
+            node[RENDER_DATA] = {}
+        for key, value in data.items():
+            if value is None:
+                continue
+            node[RENDER_DATA][key] = value
+
+    def prepare_rendering(self, asset_group, name):
+        filepath = bpy.data.filepath
+        assert filepath, "Workfile not saved. Please save the file first."
+
+        file_path = os.path.dirname(filepath)
+        file_name = os.path.basename(filepath)
+        file_name, _ = os.path.splitext(file_name)
+
+        project = get_current_project_name()
+        settings = get_project_settings(project)
+
+        render_folder = self.get_default_render_folder(settings)
+        aov_sep = self.get_aov_separator(settings)
+        ext = self.get_image_format(settings)
+        multilayer = self.get_multilayer(settings)
+
+        aov_list, custom_passes = self.set_render_passes(settings)
+
+        output_path = os.path.join(file_path, render_folder, file_name)
+
+        render_product = self.get_render_product(output_path, name)
+        aov_file_product = self.set_node_tree(
+            output_path, name, aov_sep, ext, multilayer)
+
+        # We set the render path, the format and the camera
+        bpy.context.scene.render.filepath = render_product
+        self.set_render_format(ext, multilayer)
+
+        render_settings = {
+            "render_folder": render_folder,
+            "aov_separator": aov_sep,
+            "image_format": ext,
+            "multilayer_exr": multilayer,
+            "aov_list": aov_list,
+            "custom_passes": custom_passes,
+            "render_product": render_product,
+            "aov_file_product": aov_file_product,
+        }
+
+        self.imprint_render_settings(asset_group, render_settings)
+
+    def process(self):
+        # Get Instance Container or create it if it does not exist
+        instances = bpy.data.collections.get(AVALON_INSTANCES)
+        if not instances:
+            instances = bpy.data.collections.new(name=AVALON_INSTANCES)
+            bpy.context.scene.collection.children.link(instances)
+
+        # Create instance object
+        asset = self.data["asset"]
+        subset = self.data["subset"]
+        name = plugin.asset_name(asset, subset)
+        asset_group = bpy.data.collections.new(name=name)
+
+        try:
+            instances.children.link(asset_group)
+            self.data['task'] = get_current_task_name()
+            lib.imprint(asset_group, self.data)
+
+            self.prepare_rendering(asset_group, name)
```

Comment:
Note that the instance will be created - and imprinted even if `prepare_rendering` fails due to the file not being saved. Would it make more sense to perform the "ensure file is saved" check at the beginning of `process` to stop immediately before doing anything?

Reply:
The problem is that because of Blender limitations, the file is just saved, we cannot check if there are changes (unless they are made via UI). The `try` is there to remove the instance if any other error arises (I think there was a complaint about that, that if it fails for any reason, the instance must be manually deleted).